### ISSUE TITLE
In a commonjs environment, when A is exporting B and B is extending C, A is imported, but B appears.

### ISSUE THREAD

**Issue by younkue** [monounity/karma-typescript#316]
## In a commonjs environment, when A is exporting B and B is extending C, B is imported.
* A.ts
```ts
const B = require("B");
export default class A extends B {
}
```
* B.ts
```ts
export default class B {
}
```

### Build modules A and B as commonjs.
* test.spec.ts
```ts
import A from "A";

console.log(A); // B
```
The result is printed as B, not A.

### commonjs.js are problematic files.
* src/client/commonjs.js (line: 20)
```js
if(module.exports && !module.exports.default && isExtensible(module.exports)) {
  Object.defineProperty(module.exports, "default", {
    enumerable: false, configurable: true, writable: true, value: module.exports
  });
}
```
'Require' creates 'default' for B.

However, A overwrites the default of B in the process of extending B.

```js
A.default = B.default = B
```

Suggest the source as follows:
* src/client/commonjs.js (line: 20)
```js
if(
    exports && isExtensible(exports) &&
    ((typeof exports === "function" && exports !== exports.default) || !exports.default)
) {
    if (!exports.__esModule) {
        Object.defineProperty(exports, "__esModule", {
            enumerable: false, configurable: true, writable: true, value: true
        });
    }
    Object.defineProperty(exports, "default", {
        enumerable: false, configurable: true, writable: true, value: exports
    });
}
```

**Comment by erikbarke** [comment]
All integration tests pass with your fix, I'll try to put together an integration test that breaks without it!

**Comment by erikbarke** [comment]
@younkue, I just commited an attempt at recreating your issue in e76abe0, but I can't get the test to break without your fix, could you take look at the integration test please?

**Comment by younkue** [comment]
@erikbarke 

Commonjs are used as follows.


https://github.com/younkue/karma-typescript-test

"exports.default" is OK. But karma-typescript is problematic because it is converted to "module.exports" in the umd case.

* A module
https://github.com/younkue/karma-typescript-test/blob/master/A/index.js


* B module
https://github.com/younkue/karma-typescript-test/blob/master/B/index.js


* Test Case

https://github.com/younkue/karma-typescript-test/blob/master/test/test.spec.ts

**Comment by erikbarke** [comment]
Ok, I can reproduce the errors when running your example project on my end, thanks!

**Comment by erikbarke** [comment]
Your fix is in #323 now ☺️
